File: python/guessnumber.py
"""
Guess the Number

A game in which the user must guess a number.
"""

import random

number = random.randint(1, 100) # random number between 1 and 100

print("I'm thinking of a number between 1 and 100")
while True:
    print('Take a guess')
    guess = int(input('> ')) # get the user's guess

    if guess == number:
        print('Correct! You win!')
        break
    if guess > number:
        print('Your guess is too high')
    elif guess < number:
        print('Your guess is too low')

File: java/GuessNumber.java
import java.util.*;

/**
 * @name GuessNumber
 * @version 0.0.1
 */

public class GuessNumber {
  public static void main(String[] args) {
    int number = (int)(Math.random()*100); // make a random integer between 1 and 100
       
    int guess = 101; // so that the guess cannot be the answer
    Scanner scanner = new Scanner(System.in);
       
    System.out.println("Guess a number between 1 and 100");
        
    while (guess != number) {
      guess = scanner.nextInt();

      if (guess < number) {
        System.out.println(guess + " is too small, try again.");
      } else if (guess > number) {
        System.out.println(guess + " is too big, try again.");
      } else {
        break;
      }
    }
    System.out.println("Correct!");
  }
}
